File: java/v1/src/test/java/com/api/v1/rent/AllRentsRetrievalTest.java
package com.api.v1.rent;

import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestMethodOrder;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.reactive.server.WebTestClient;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class AllRentsRetrievalTest {

    @Autowired
    WebTestClient webTestClient;

    @Test
    @Order(1)
    void testSuccessfulAllRentsRetrieval() {
        webTestClient
                .get()
                .uri("api/v1/rents")
                .exchange()
                .expectStatus()
                .is2xxSuccessful();
    }

}

File: kotlin/v2/src/test/kotlin/com/api/v2/rent/AllRentsRetrievalTest.kt
package com.api.v2.rent

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.web.reactive.server.WebTestClient

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
private class AllRentsRetrievalTest {

    @Autowired
    lateinit var webTestClient: WebTestClient

    @Test
    fun testIfRntIsEmpty() {
        webTestClient
            .get()
            .uri("api/v2/rents")
            .exchange()
            .expectStatus()
            .is2xxSuccessful()
    }

}